### ISSUE TITLE
Object Browser displays incorrect objects/members, and is painfully slow

### ISSUE THREAD

**Issue by verelpode** [dotnet/project-system#4083]
Hello, the Object Browser window is very useful -- I love it and I thank everyone who ever worked on it, because I use it multiple times every day I work.  However, some reliability and speed improvements would be very much appreciated.  Sometimes it displays incorrect objects/members (see screenshot).  It replaces the correct object/member with some other random object/member, seemingly chosen at random, as if the internal data structures are corrupted.  VS 15.8.5.

Furthermore, it is unfortunately slow to the point of being frustrating to use/wait.  Searching is slow, and expanding "Desired Types" is extremely slow.  When I change "Browse:" to "Universal Windows" then search etc, it seems even slower than when I have it set to browse ".NET Framework 4.7.2".  Unsurprisingly, the speed is worsened even further when you search "All Components".   I suggest using indexing and caching techniques to improve the speed.  Maybe even pre-load all of the data into an SQLite database, then search the database instead of searching the original assemblies.

See also [Dev Community #76369](https://developercommunity.visualstudio.com/content/problem/76369/object-browser-blocking.html) -- another person who described the painful slowness of Object Browser.

![object browser bug](https://user-images.githubusercontent.com/32857007/46488173-3df8b780-c802-11e8-8725-4c799a4df25e.png)

**Comment by verelpode** [comment]
Example of the speed problem:  I chose `Browse: All Components` then I searched for the word "**updated**".  When the search completed, I immediately clicked the button to clear the search, which caused another delay.  The total time was **21 seconds**!  During these 21 seconds, Visual Studio was totally unresponsive to mouse clicks etc (the search is not async).

Next I searched for "**visual**" and it again took around 20 seconds, but when I searched for "**xaml**", it took half the time -- around 10 seconds (including time to also clear the search).

Next I tried opening "**Derived Types**" for `Windows.UI.Xaml.UIElement` and it took a shocking 1 minute and 38 seconds!
